bot/cogs/tickets.py

```diff
@@ -0,0 +1,405 @@
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import TYPE_CHECKING, NamedTuple, Optional, Union
+
+import asyncpg
+import discord
+from discord.ext import commands
+from discord.utils import format_dt, utcnow
+from libs.tickets.structs import ReservedTags, StatusChecklist, TicketThread
+from libs.tickets.utils import get_cached_thread, get_partial_ticket
+from libs.utils.embeds import Embed, LoggingEmbed
+
+from .config import GuildWebhookDispatcher
+
+if TYPE_CHECKING:
+    from libs.utils import RoboContext
+    from rodhaj import Rodhaj
+
+
+STAFF_ROLE = 1184257456419913798
+
+
+def is_ticket_or_dm():
+    def pred(ctx: RoboContext) -> bool:
+        partial_config = ctx.partial_config
+        return (
+            isinstance(ctx.channel, discord.Thread)
+            and partial_config is not None
+            and ctx.channel.parent_id == partial_config.ticket_channel_id
+        ) or ctx.guild is None
+
+    return commands.check(pred)
+
+
+class TicketOutput(NamedTuple):
+    status: bool
+    ticket: discord.channel.ThreadWithMessage
+    msg: str
+
+
+class ClosedEmbed(discord.Embed):
+    def __init__(self, **kwargs):
+        kwargs.setdefault("color", discord.Color.from_rgb(138, 255, 157))
+        kwargs.setdefault("title", "\U00002705 Ticket Closed")
```

Comment:
Why escape the emoji instead of just including it in the string? Like `"✅ Ticket Closed"`

Reply:
`kwargs.setdefault("title", "\U00002705 Ticket Closed") # U+2705 White Heavy Check Mark`

Reply:
Fixed in commit adee012304df96d8a492c485c54dedcdd8a879bb